File: src/main/scala/hohserg/endothermic/example/java/ReconstructExample.java
package hohserg.endothermic.example.java;

import hohserg.endothermic.quad.immutable.UnpackedQuad;
import net.minecraft.client.renderer.block.model.BakedQuad;

import java.util.function.Function;

public class ReconstructExample implements Function<BakedQuad, BakedQuad> {

    @Override
    public BakedQuad apply(BakedQuad quad) {
        return UnpackedQuad.apply(quad)
                .reconstructBuilder()//no allocation
                .v1_x(0.5f)//updated x of first vertex
                .build()
                .toBakedQuad();
    }
}

File: src/main/scala/hohserg/endothermic/example/scala/ReconstructExample.scala
package hohserg.endothermic.example.scala

import hohserg.endothermic.quad.immutable.LazyUnpackedQuad
import net.minecraft.client.renderer.block.model.BakedQuad

class ReconstructExample extends (BakedQuad => BakedQuad) {
  override def apply(v1: BakedQuad): BakedQuad =
    LazyUnpackedQuad(v1)
      .reconstruct(v1_x = 0.5f)
      .toBakedQuad
}
